src/arithmetic/canonical.cc

```diff
@@ -526,6 +558,10 @@ class Canonical::Internal : public IRMutator {
     if (pair.size() == 0) {
       int64_t value = GetConstIntValue(v);
       auto n = make_node<ComExprNode>();
+      // TODO(derisavi) : fix this bug. The following can be done only for Euclidean division/mod.
+      //  Therefore, it's only valid when truncated division/mod is equivalent to Euclidean one,
+      //  that is, if and only if a and v are
+      //  both negative or both positive or a is divisible by v.
```

Comment:
so, to fix, we need to first define tvm's mod operator where a and v have contrary sign?

Reply:
I changed the comment a bit more to make it more understandable.

AFAIK, TVM's semantics of div and mod is known for all signs of a and v; it's the same semantics used in C++11, i.e., div result is truncated toward zero and a%v = a - v*(a/v)

Reply:
Thanks, so in my understanding, the current implement is *incorrect* given "a and v are of different signs and a is not divisible by v", right? Does it require large effort to fix?

Reply:
That's right. It's incorrect now. 

Fixing it should be straightforward. It requires new unit testcases, and possibly, reviewing some of the existing ones. I think it should be done in a separate PR though.

Reply:
yeah it makes sense, suggest to mark it FIXME then.

Reply:
done